### ISSUE TITLE
MAPL develop failure with GNU debug

### ISSUE THREAD

**Issue by mathomp4** [GEOS-ESM/MAPL#1276]
MAPL `develop` failed last night running with GNU debug. The error message was:
```
At line 1531 of file /discover/nobackup/mathomp4/SystemTests/builds/AGCM_GNUMAPLDEV/CURRENT/GEOSgcm/src/Shared/@MAPL/generic/MAPL_Generic.F90
Fortran runtime error: Recursive call to nonrecursive procedure 'initialize_children_and_couplers'
#0  0x2aaabbac7abd in initialize_children_and_couplers
        at /discover/nobackup/mathomp4/SystemTests/builds/AGCM_GNUMAPLDEV/CURRENT/GEOSgcm/src/Shared/@MAPL/generic/MAPL_Generic.F90:1531
#1  0x2aaabbad2bce in __mapl_genericmod_MOD_mapl_genericinitialize
        at /discover/nobackup/mathomp4/SystemTests/builds/AGCM_GNUMAPLDEV/CURRENT/GEOSgcm/src/Shared/@MAPL/generic/MAPL_Generic.F90:1229
...
```

This seems to be the same procedure as pointed to in #1275. As this is a bit more informative, I'll try out the suggested solution.

**Comment by mathomp4** [comment]
Possibly related to #1276

**Comment by tclune** [comment]
This should be fixed in my PR from last night.

**Comment by mathomp4** [comment]
> This should be fixed in my PR from last night.

Indeed, I'm testing that now. My hope is it might fix #1275 as well. But I'll start with GNU as it's seems more concrete.
